pkg/managed/serviceaccount_test.go

```diff
@@ -0,0 +1,237 @@
+	return map[string]*util.ManagedClusterConnection{
+		"local":     testutil.GetManagedClusterConnection("local"),
+		"cluster-1": testutil.GetManagedClusterConnection("cluster-1"),
+	}
+}
+
+// createAppNamespaces creates the namespaces needed for a application binding.
+func createAppNamespaces(t *testing.T, managedConnections map[string]*util.ManagedClusterConnection) {
+	var ns = corev1.Namespace{}
+
+	clusterConnection := managedConnections["local"]
+	ns.Name = "ns-test1"
+	_, err := clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+	ns.Name = "ns-test2"
+	_, err = clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+
+	clusterConnection = managedConnections["cluster-1"]
+	ns.Name = "ns-test3"
+	_, err = clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+}
+
+// createVMINamespaces creates the namespaces needed for a VMI binding.
+func createVMINamespaces(t *testing.T, managedConnections map[string]*util.ManagedClusterConnection) {
+	var ns = corev1.Namespace{}
+
+	clusterConnection := managedConnections["local"]
+	ns.Name = constants.LoggingNamespace
+	_, err := clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+	ns.Name = constants.MonitoringNamespace
+	_, err = clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+
+	clusterConnection = managedConnections["cluster-1"]
+	ns.Name = constants.LoggingNamespace
+	_, err = clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+	ns.Name = constants.MonitoringNamespace
+	_, err = clusterConnection.KubeClient.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(fmt.Sprintf("can't create namespace %s: %v", ns.Name, err))
+	}
+}
+
+// assertCreateAppServiceAccounts asserts that the given service account for an application binding has
+// the expected values.
+func assertCreateAppServiceAccounts(t *testing.T, sa *corev1.ServiceAccount, clusterName string) {
+	assert := assert.New(t)
+
+	labels := util.GetManagedLabelsNoBinding(clusterName)
+
+	assert.Equal(constants.VerrazzanoNamespace, sa.Name, "namespace %s service account name name was not as expected", sa.Namespace)
```

Comment:
shouldn't param 2 be sa.Namespace?

Reply:
fixed.  I changed the first arg to be util.GetServiceAccountNameForSystem() which is the same as constants.VerrazzanoNamespace.